HealthVault/Classes/Clients/MHVPersonClient.m

```diff
@@ -70,56 +130,274 @@ - (void)getAuthorizedPeopleWithCompletion:(void(^)(NSArray<MHVPersonInfo *> *_Nu
+             
+             return;
+         }
+         
+         if (peopleResult.persons.count < 1)
+         {
+             completion(nil, [NSError error:[NSError MHVUnknownError] withDescription:@"The GetAuthorizedPeople response has no authorized people."]);
+             
+             return;
+         }
+         
+         completion(peopleResult, nil);
+     }];
+}
+
+- (NSString *)bodyForSetApplicationSettings:(MHVApplicationSettings *)applicationSettings
+{
+    XWriter *writer = [[XWriter alloc] initWithBufferSize:2048];
+    
+    [writer writeStartElement:@"info"];
+
+    if ([self isValidObject:applicationSettings])
     {
-        if (error)
-        {
-            completion(nil, error);
-            
-            return;
-        }
-        
-        MHVGetAuthorizedPeopleResult *peopleResult = (MHVGetAuthorizedPeopleResult *)[XSerializer newFromString:response.infoXml withRoot:@"info" asClass:[MHVGetAuthorizedPeopleResult class]];
-       
-        if (!peopleResult)
-        {
-            completion(nil, [NSError error:[NSError MHVUnknownError] withDescription:@"The GetAuthorizedPeople response is invalid."]);
-            
-            return;
-        }
-        
-        if (peopleResult.persons.count < 1)
-        {
-            completion(nil, [NSError error:[NSError MHVUnknownError] withDescription:@"The GetAuthorizedPeople response has no authorized people."]);
-            
-            return;
-        }
-        
-        completion(peopleResult.persons, nil);
-        
-    }];
+        [XSerializer serialize:applicationSettings withRoot:@"app-settings" toWriter:writer];
+    }
+
+    [writer writeEndElement];
+
+    return [writer newXmlString];
 }
 
-- (void)getAuthorizedPeopleWithSettings:(MHVGetAuthorizedPeopleSettings *_Nonnull)settings
-                             completion:(void(^)(NSArray<MHVPersonInfo *> *_Nullable personInfos, NSError *_Nullable error))completion
+- (NSString *)bodyForGetAuthorizedPeopleSettings:(MHVGetAuthorizedPeopleSettings *)authorizedPeopleSettings
 {
+    XWriter *writer = [[XWriter alloc] initWithBufferSize:2048];
+    
+    [writer writeStartElement:@"info"];
+    
+    if ([self isValidObject:authorizedPeopleSettings])
+    {
+        [XSerializer serialize:authorizedPeopleSettings withRoot:@"parameters" toWriter:writer];
+    }
     
+    [writer writeEndElement];
+    
+    return [writer newXmlString];
 }
 
-- (void)getPersonInfoWithCompletion:(void(^)(MHVPersonInfo *_Nullable person, NSError *_Nullable error))completion
+- (NSString *)bodyForGetAuthorizedRecords:(NSArray<NSUUID *> *)recordIds
 {
+    XWriter *writer = [[XWriter alloc] initWithBufferSize:2048];
+    
+    [writer writeStartElement:@"info"];
+    
+    for (NSUUID *uuid in recordIds)
+    {
+        [XSerializer serialize:[[MHVString alloc] initWith:uuid.UUIDString] withRoot:@"id" toWriter:writer];
+    }
+
+    [writer writeEndElement];
     
+    return [writer newXmlString];
```

Comment:
I think we need to return nil here (or some other error check) if the string is not valid XML. This code, allows the user to make the set settings call with invalid XML. This method writes an empty info object, and the call will complete successfully even though we didn't actually save anything.